### PR TITLE
feat/matic portfolio

### PR DESCRIPTION
**Pull request by garyunwin42**
Add portfolio on polygon chain

## Scope

https://linear.app/dopex/issue/DOP-348/add-portfolio-on-the-polygon-chain

## Screenshots
n/a

## How to Test
Go to /portfolio and see positions from polygon chain

### CODE REVIEW

**Review by witherblock — CHANGES_REQUESTED** [review]
Portfolio should fetch data for all chains not which chain the user is in.

Also update the template description of the PR correctly.

File: apps/dapp/src/store/Portfolio/index.ts [hunk 1]
```diff
@@ -299,109 +301,178 @@ export const createPortfolioSlice: StateCreator<
...
+        queryKey: ['getSsovUserData'],
+        queryFn: async () =>
+          request(
+            DOPEX_POLYGON_SSOV_SUBGRAPH_API_URL,
+            getSsovUserDataDocument,
+            {
+              user: accountAddress.toLowerCase(),
+            }
+          ),
+      }),
```
Comment (hibou-de-nuit) [inline apps/dapp/src/store/Portfolio/index.ts:322]: Let's query based on chainId, add a mapping like this: 
```ts
const chainIdToQuery: Record<string | number, string> = {
   [chainId]: API_URL,
  ...
}
```

Comment (hibou-de-nuit) [reply apps/dapp/src/store/Portfolio/index.ts:322]: cc. [here](https://github.com/dopex-io/elvarg/pull/601/files#diff-ca1754bc517e40750e49c2d67e0663d96f63b832b6ca3d528b3a446b6815b2e8R372), [here](https://github.com/dopex-io/elvarg/pull/601/files#diff-ca1754bc517e40750e49c2d67e0663d96f63b832b6ca3d528b3a446b6815b2e8R375), [here](https://github.com/dopex-io/elvarg/pull/601/files#diff-ca1754bc517e40750e49c2d67e0663d96f63b832b6ca3d528b3a446b6815b2e8R381), and several following lines querying both polygon as well as arbitrum.

**Review by hibou-de-nuit — CHANGES_REQUESTED** [review]
The portfolio page is stuck in infinite loading for both chains. Besides that and some improvements, LGTM.

File: apps/dapp/src/store/Portfolio/index.ts (hunk 1 above)
Comment (hibou-de-nuit) [reply apps/dapp/src/store/Portfolio/index.ts:322]: Noted witherblock's review. Please ignore.

**Comment by garyunwin42** [comment]
> The portfolio page is stuck in infinite loading for both chains. Besides that and some improvements, LGTM.

where are you seeing the infinite loading? it worked for me and here's the preview: https://dapp-git-feat-matic-portfolio-dopex-io.vercel.app/portfolio

**Review by witherblock — DISMISSED** [review]
Fix PR description

**Comment by hibou-de-nuit** [comment]
> > The portfolio page is stuck in infinite loading for both chains. Besides that and some improvements, LGTM.
> 
> where are you seeing the infinite loading? it worked for me and here's the preview: https://dapp-git-feat-matic-portfolio-dopex-io.vercel.app/portfolio

Same issue in the preview link. It is likely that the straddlesv2__factory is not accounted for in the portfolio

**Review by hibou-de-nuit — APPROVED** [review]
New changes LGTM. 

Some other issues were noted. 
- For a straddles deposit, the 'open' button takes us to a broken page. the URL param ```?epoch=###``` breaks the page and doesn't load contract data.
- When connected to Polygon chain, SSOV positions from the arb chain do not appear.

We're good to merge once these have been addressed.

File: apps/dapp/src/store/Portfolio/index.ts [hunk 2]
```diff
@@ -228,15 +259,33 @@ export const createPortfolioSlice: StateCreator<
...
+        const vault = AtlanticStraddleV2__factory.connect(
+          vaultAddress,
+          provider
+        );
+        epoch = userPosition.epoch;
+
+        responses = await Promise.all([
+          vault.symbol(),
+          vault.epochStatus(epoch),
+          vault.paused,
```
[the remark this answers is not in the document]
Comment (garyunwin42) [reply apps/dapp/src/store/Portfolio/index.ts:277]: weird, it worked for me
image

[the remark this answers is not in the document]
Comment (garyunwin42) [reply apps/dapp/src/store/Portfolio/index.ts:277]: open from portfolio worked for me too... 😂 
image

[the remark this answers is not in the document]
Comment (hibou-de-nuit) [reply apps/dapp/src/store/Portfolio/index.ts]: Oddly still seems to be the case for me. I switched to polygon chain on the portfolio, refreshed, and get the result below. 
Screenshot 2023-07-10
Screenshot 2023-07-10
Screenshot 2023-07-10

[the remark this answers is not in the document]
Comment (garyunwin42) [reply apps/dapp/src/store/Portfolio/index.ts]: what's your address? lemme try to reproduce with yours

[the remark this answers is not in the document]
Comment (garyunwin42) [reply apps/dapp/src/store/Portfolio/index.ts]: i refreshed but it worked for me
image

**Pull request merged**